### ISSUE TITLE
Make nest depends on Redic instead of Redis

### ISSUE THREAD

**Issue by phuongnd08** [soveran/nest#5]
I think Redic seem to be the better option. Per my experience, Redic doesn't yield issue when being wrapped inside an EventMachine block.

**Comment by deric** [comment]
Right now `nest` is calling `redis` with an extra `&block` argument ([L33](https://github.com/soveran/nest/blob/master/lib/nest.rb#L33)):
```ruby
redis.send(meth, self, *args, &block)
```
Passed block `redis` uses in some methods, e.g. `subscribe` in [redis.rb](https://github.com/redis/redis-rb/blob/7c4a95413009cefa0c74d8d320f1ae90a1c953c2/lib/redis.rb#L1974):

```ruby
 def subscribe(*channels, &block)
    synchronize do |client|
      _subscription(:subscribe, channels, block)
    end
  end
```
In this case `&block` is called asynchronously, which is not supported in `redic`.  In order to maintain same functionality in `nest` we would have to implement methods like `subscribe` , `psubscribe`, `scan_each`, `hscan_each`, etc. Or extend `redic` to support this, but it might be against the design principle of `redic`. 

What do you think?

**Comment by soveran** [comment]
How are you using Nest? I just checked that Ohm 1.4.x uses Nest 1.x, so we can make Nest use Redic and release it as 2.0, breaking any compatibility if necessary. What do you think about that?

**Comment by deric** [comment]
Sounds like a good idea. I'm using Nest along with Ohm 2.2.

**Comment by soveran** [comment]
Done, it is available in Nest 2.0.0. Thanks!
